src/engraving/libmscore/element.cpp

```diff
@@ -1377,18 +1377,18 @@ void paintElements(mu::draw::Painter& painter, const QList<Element*>& elements)
         if (e1->z() == e2->z()) {
             if (e1->selected()) {
                 return false;
+            } else if (e1->visible()) {
+                return false;
             } else if (e2->selected()) {
                 return true;
-            } else if (!e1->visible()) {
+            } else if (e1->visible()) {
                 return true;
-            } else if (!e2->visible()) {
-                return false;
             }
 
             return e1->track() > e2->track();
         }
 
-        return e1->z() <= e2->z();
+        return e1->z() < e2->z();
```

Comment:
unrelated change but without it was getting lots of asserts about the predicate being passed to std::sort( ) was invalid, as it didn't consistently return a < b when b > a.